tests/integration/telemetry/stackdriver/stackdriver_filter_audit_test.go

```diff
@@ -0,0 +1,108 @@
+// +build integ
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackdriver
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+)
+
+const serverAuditLogEntry = "testdata/server_audit_log.json.tmpl"
+
+// TestStackdriverAuditLogging testing Authz Policy can config stackdriver with audit policy
+func TestStackdriverAuditLogging(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(ctx framework.TestContext) {
+
+			ns := getEchoNamespaceInstance().Name()
+			args := map[string]string{
+				"Namespace": ns,
+			}
+			policies := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, "testdata/v1beta1-audit-authorization-policy.yaml.tmpl"))
+			ctx.Config().ApplyYAMLOrFail(t, ns, policies...)
+			defer ctx.Config().DeleteYAMLOrFail(t, ns, policies...)
+			t.Logf("Audit policy deployed to namespace %v", ns)
+
+			sendTrafficForAudit(t)
+			t.Logf("Traffic sent to namespace %v", ns)
+			for _, cl := range ctx.Clusters() {
+				scopes.Framework.Infof("Validating Audit policy and Telemetry for Cluster %v", cl)
+
+				retry.UntilSuccessOrFail(t, func() error {
+					clName := cl.Name()
+					t.Logf("Collect Audit Log for cluster %v", clName)
+					if err := validateLogs(t, serverAuditLogEntry, clName); err != nil {
```

Comment:
I'm very familiar with the telemetry test but does this make sure it is checking the audit log entry instead of the normal access log entry? @gargnupur  @bianpengyuan

Reply:
The reason it should work is because audit log entry logs much less data then the normal access log entry, hence if it was validating normal access log entry, it will fail.

To make it full proof, if needed , we can add logName in verification.

Reply:
I see, I think we should verify the logName if it's not too hard to add, or at least put a comment about this if we don't do it now.